src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1360,6 +1360,26 @@ MavlinkReceiver::handle_message_battery_status(mavlink_message_t *msg)
 	battery_status.cell_count = cell_count;
 	battery_status.connected = true;
 
+	// Get the battery level thresholds.
+	float bat_emergency_thr;
+	float bat_crit_thr;
+	float bat_low_thr;
+	param_get(param_find("BAT_EMERGEN_THR"), &bat_emergency_thr);
```

Comment:
param_find searches the full param set each time. You could do this once and then hold the param_t handle.